### PR TITLE
Update to avvio@7.0.0-alpha.0

### PR DESCRIPTION
**Pull request by mcollina**
Update to avvio@7, implementing `await app.register()` and `await app.after()`.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
@jsumners I think we should plan the release for v3 with some caution. I'm thinking to have some -alpha releases and a few release candidates. We can then update the modules. I would lean to do a semver-major change everywhere because I think we should drop old Node.js versions anyway.

**Comment by jsumners** [comment]
I agree. I just want to make sure we don't release a _final_ v3 with release candidate tagged dependencies.

File: docs/Server.md [hunk 1]
```diff
@@ -425,6 +425,26 @@ fastify
...
+In case `after()` is called without a function, it returns a `Promise`:
+
+```js
+fastify.register(async (instance, opts) => {
+  console.log('Current plugin')
+})
+
+await fastify.after()
+  console.log('After current plugin')
+})
```
Comment (delvedor) [inline docs/Server.md]: Leftover :)

**Comment by mcollina** [comment]
I've added a test for `await app.register()` and found a non-blocking bug.

@delvedor which kind of doc would you like to see? I struggled a bit in finding a good place where to add more.

I would like to land this ASAP so we can unblock all the other PRs.

**Pull request merged**